### PR TITLE
[5.1] Added function to manage multiple SQL where clauses in array to get results

### PR DESCRIPTION
**Pull request by srmklive**
Created method to add multiple SQL where clauses inside an multi-dimensional array instead of chaining multiple where function calls for eloquent.

For example:

$users = App\User::where('status','=','1')->where('subscribed','=','1')->get();

changes to

$users = App\User::whereAll([
['status','=','1'],
['subscribed','=','1'],
])->get();

### CODE REVIEW

File: src/Illuminate/Database/Eloquent/Builder.php [hunk 1]
```diff
@@ -563,6 +563,32 @@ public function orWhere($column, $operator = null, $value = null)
     }
 
     /**
+     * Adds multiple where clauses to the query
+     *
+     * @param array $conditions
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Builder.php]: cs

File: src/Illuminate/Database/Query/Builder.php [hunk 2]
```diff
@@ -522,6 +522,32 @@ public function orWhere($column, $operator = null, $value = null)
     }
 
     /**
+     * Adds multiple where clauses to the query
+     *
+     * @param array $conditions
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/Builder.php]: cs

File: src/Illuminate/Database/Query/Builder.php [hunk 3]
```diff
@@ -522,6 +522,32 @@ public function orWhere($column, $operator = null, $value = null)
...
+     * @return $this
+     */
+    public function whereAll($conditions)
+    {
+        foreach ($conditions as $condition) {
+            if (sizeof($condition) == 2) {
+                @list($column, $value) = $condition;
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/Builder.php]: bad idea to silence errors

File: src/Illuminate/Database/Query/Builder.php [hunk 4]
```diff
@@ -522,6 +522,32 @@ public function orWhere($column, $operator = null, $value = null)
...
+            } else {
+                @list($column, $operator, $value, $clause) = $condition;
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/Builder.php]: bad idea to silence errors

Comment (srmklive) [inline src/Illuminate/Database/Query/Builder.php]: @GrahamCampbell Can you describe how i should go about doing it?

**Comment by acasar** [comment]
@srmklive Normal where already supports something similar:

```php
$users = App\User::where([
    'status' => '1'
    'subscribed' => '1',
])->get();
```

**Comment by srmklive** [comment]
@acasar yes but it only uses = operator by default.

**Comment by taylorotwell** [comment]
Going to hold off on this.

**Pull request closed without merging**